CoreGraphFile/src/au/gov/asd/tac/constellation/graph/file/open/RecentFiles.java

```diff
@@ -153,6 +171,16 @@ protected static List<HistoryItem> getRecentFiles() {
      * @return list of recent files
      */
     public static List<HistoryItem> getUniqueRecentFiles() {
+        if (historyReady.getCount() != 0) {
+            try {
+                LOGGER.log(Level.WARNING, ">> Timing issue encountered: Recent Files data is being accessed before it has been initialised <<");
+                if (!historyReady.await(300, TimeUnit.SECONDS)) {
+                    LOGGER.log(Level.WARNING, ">> Recent Files did not initialise within 5 minutes <<", new Exception(NotifyDisplayer.BLOCK_POPUP_FLAG + "WARNING: Recent Files data did not initialise within a reasonable time"));
+                }
+            } catch (final InterruptedException ex) { // NOSONAR
+                LOGGER.log(Level.SEVERE, ex.toString(), ex);
```

Comment:
1) are you able to add a thread interrupt here?
2) instead of `ex.toString`, can you use `ex.getLocalisedMessage()`?

Reply:
1. Yes
2. No. I prefer to use the toString() function because it contains more information, and __*includes*__ getLocalizedMessage()